torch/csrc/jit/pybind_utils.h

```diff
@@ -88,82 +97,146 @@ inline c10::optional<TypePtr> unifyOrInitializeType(
   return unifyTypes(accum, unify);
 }
 
-inline TypedIValue toTypedIValue(py::handle input) {
+MatchTypeReturn tryToInferContainerType(py::handle input);
+
+// Try to infer the type of a Python object
+// The type cannot be inferred if:
+//   input is a None
+//   input is an empty container (list, dict, tuple)
+//   input is an empty container
+//   input is an dict with more than one type of key or value
+//   input is an list with more than one type of element
```

Comment:
Shouldn't it be - `input is a list with elements types that cannot be unified` ?